<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GADialog</class>
 <widget class="QDialog" name="GADialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>443</width>
    <height>515</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Genetic Algorithm</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_12">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_11">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Do you want to get algorithm settings from file?</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="yes">
             <property name="text">
              <string>Yes</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="no">
             <property name="text">
              <string>No</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>File with algorithm settings:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="algconfname"/>
           </item>
           <item>
            <widget class="QToolButton" name="toolButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Algorithm settings(not from file):</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Number of populations:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="popNum">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>30</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>crossPercent:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="cp">
               <item>
                <widget class="QDoubleSpinBox" name="cp1">
                 <property name="decimals">
                  <number>2</number>
                 </property>
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.700000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="cp2">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.800000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="cp3">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.900000000000000</double>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Pcross:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="pc">
               <item>
                <widget class="QDoubleSpinBox" name="pc1">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.500000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="pc2">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.750000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="pc3">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>1.000000000000000</double>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Mutation Percent:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="mp">
               <item>
                <widget class="QDoubleSpinBox" name="mp1">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.700000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="mp2">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.800000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="mp3">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.900000000000000</double>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Pmut:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="pm">
               <item>
                <widget class="QDoubleSpinBox" name="pm1">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.500000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="pm2">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.750000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="pm3">
                 <property name="maximum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.900000000000000</double>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item alignment="Qt::AlignHCenter">
            <widget class="QCheckBox" name="use_metamodels">
             <property name="text">
              <string>Use metamodels</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Metamodel:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <item>
                <property name="text">
                 <string>Averaging</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>KNearestNeighbours</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Neural Network</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Support Vector Regression</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Linear Regression</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Random</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Population control percent:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="popControl">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="minimum">
                <number>3</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>Run</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="metamodels_button">
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/pics/pics/chart_line.png</normaloff>:/pics/pics/chart_line.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>496</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>metamodels_button</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>draw_metamodels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>244</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>use_metamodels</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>use_metamodels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>48</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>algconfname</sender>
   <signal>textChanged(QString)</signal>
   <receiver>GADialog</receiver>
   <slot>loadAlgConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>384</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>openDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>289</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>metamodels_button</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>showMetamodelsRes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yes</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>pressYes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>334</x>
     <y>9</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>no</sender>
   <signal>clicked()</signal>
   <receiver>GADialog</receiver>
   <slot>pressNo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>run()</slot>
  <slot>draw_metamodels()</slot>
  <slot>use_metamodels()</slot>
  <slot>loadAlgConf()</slot>
  <slot>openDialog()</slot>
  <slot>change_metamodel()</slot>
  <slot>changePCPersent()</slot>
  <slot>showMetamodelsRes()</slot>
  <slot>pressYes()</slot>
  <slot>pressNo()</slot>
 </slots>
</ui>
